li-utils/src/main/javaPegasus/com/linkedin/common/urn/NotebookUrn.java

```diff
@@ -0,0 +1,71 @@
+package com.linkedin.common.urn;
+
+import com.linkedin.data.template.Custom;
+import com.linkedin.data.template.DirectCoercer;
+import com.linkedin.data.template.TemplateOutputCastException;
+import java.net.URISyntaxException;
+
+
+public class NotebookUrn extends Urn {
```

Comment:
This is not needed. We just use Urn objects for all references to Urn

Reply:
It should be fine to have this class.

Reply:
I feel we should prefer strong typed urn which could resolve unexpected error. Like in DataPlatformInstance.pdl we have a field `instance: optional Urn` but we could not easily know what the expected entity type is and how many entityKeys it expects

Reply:
We aim to minimize the amount of code added to the code base as we add more entities. Defining the key aspect defines the structure of the urn. By adding this additional class, we replicate that logic and introduce a possibility of these two being out of sync. Also, I believe this is only used by the test class?

Reply:
I think I did not make the conversation clear, sorry about that. What I was trying to say is to have some clear definition somewhere for any type of urns, like what the entity type should be and how many entity key there will be and what each of the entity key's meaning is

Reply:
So when referring to other urns, we recommend adding a relationship annotation which defines the set of entity types this urn can be. For each entity type, the structure of the urn is defined by it's key aspect.

Reply:
Do you mean ideally the `instance: optional Urn` field in `DataPlatformInstance.pdl` should have a relationship annotation pointing to the entity type?

Reply:
Yes. We will add those in as we strengthen our urn validation logic. This will be in this week!